subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -264,13 +303,14 @@ void pe_report_error(const struct device *dev, const enum pe_error e,
 	 *     response.
 	 */
 	/* All error types besides transmit errors are Protocol Errors. */
-	if ((e != ERR_XMIT && atomic_test_bit(pe->flags, PE_FLAGS_INTERRUPTIBLE_AMS) == false) ||
-	    e == ERR_XMIT ||
-	    (atomic_test_bit(pe->flags, PE_FLAGS_EXPLICIT_CONTRACT) == false &&
-	     type == PD_PACKET_SOP)) {
+	if ((e != ERR_XMIT) && pe_soft_reset_is_required(dev, type)) {
 		policy_notify(dev, PROTOCOL_ERROR);
 		pe_send_soft_reset(dev, type);
 	}
+	/* Transmit error */
+	else if (e == ERR_XMIT) {
+		atomic_set_bit(pe->flags, PE_FLAGS_MSG_XMIT_ERROR);
```

Comment:
Where does the soft reset occur for xmit errors now?  Just left to each PE state's discretion?

Reply:
Yes it's left up to the PE state's discretion.